src/test/java/org/synyx/urlaubsverwaltung/settings/web/SettingsControllerTest.java

```diff
@@ -1,17 +1,304 @@
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.flash;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+@RunWith(MockitoJUnitRunner.class)
 public class SettingsControllerTest {
 
+    private SettingsController sut;
+
+    private static final String OATUH_REDIRECT_REL = "/google-api-handshake";
+    private static final String ERRORS_ATTRIBUTE = "errors";
+    private static final String OAUTH_ERROR_ATTRIBUTE = "oautherrors";
+    private static final String OAUTH_ERROR_VALUE = "some-error";
+
+    private static final CalendarProvider SOME_CALENDAR_PROVIDER = new SomeCalendarProvider();
+    private static final CalendarProvider ANOTHER_CALENDAR_PROVIDER = new AnotherCalendarProvider();
+    private static final List<CalendarProvider> CALENDAR_PROVIDER_LIST =
+        Arrays.asList(SOME_CALENDAR_PROVIDER, ANOTHER_CALENDAR_PROVIDER);
+
+    private static final String SOME_GOOGLE_REFRESH_TOKEN = "0815-4711-242";
+
+    @Mock
+    private SettingsService settingsService;
+
+    @Mock
+    private MailService mailService;
+
+    @Mock
+    private SettingsValidator settingsValidator;
+
+    @Before
+    public void setUp() {
+
+        sut = new SettingsController(settingsService, CALENDAR_PROVIDER_LIST, mailService, settingsValidator);
+    }
+
     @Test
     public void getAuthorizedRedirectUrl() {
-        SettingsController cut = new SettingsController(null, null, null, null);
 
-        String actual = cut.getAuthorizedRedirectUrl("http://localhost:8080/web/settings", ControllerConstants.OATUH_REDIRECT_REL);
-        String expected = "http://localhost:8080/web" + ControllerConstants.OATUH_REDIRECT_REL;
-        Assert.assertEquals(expected, actual);
+        String actual = sut.getAuthorizedRedirectUrl("http://localhost:8080/web/settings", OATUH_REDIRECT_REL);
+        String expected = "http://localhost:8080/web" + OATUH_REDIRECT_REL;
+        assertEquals(expected, actual);
```

Comment:
Hey, wir hatten uns im UV Team darauf geeinigt Assertj als assertion library zu verwenden, das wäre dufte wenn du das hier auch machen könntest :) 

https://joel-costigliola.github.io/assertj/
https://www.baeldung.com/introduction-to-assertj